src/Renci.SshNet/ScpClient.cs

```diff
@@ -747,7 +784,14 @@ private void InternalDownload(IChannelSession channel, Stream input, FileSystemI
 
                     directoryCounter--;
 
-                    currentDirectoryFullName = new DirectoryInfo(currentDirectoryFullName).Parent.FullName;
+                    var currentDirectoryParent = new DirectoryInfo(currentDirectoryFullName).Parent;
+
+                    if (currentDirectoryParent is null)
+                    {
+                        break;
+                    }
+
+                    currentDirectoryFullName = currentDirectoryParent.FullName;
```

Comment:
I think this actually fixes a NullReferenceException when downloading a directory to C:\\. It would be more explicit if the `directoryCounter == 0` check were first (I can't add as a suggestion):

```c#
directoryCounter--;

if (directoryCounter == 0)
{
    break;
}

var currentDirectoryParent = new DirectoryInfo(currentDirectoryFullName).Parent;

Debug.Assert(currentDirectoryParent is not null, $"Should be {directoryCounter} levels deeper than {startDirectoryFullName}.");

currentDirectoryFullName = currentDirectoryParent.FullName;

continue;
```

Reply:
@Rob-Hague I just pushed another change to simply disable the nullable warnings on old frameworks since these kind of false positives would become very annoying when migrating more of the codebase to nullable.